src/lib.rs

```diff
@@ -729,6 +729,10 @@ impl SystemHandle {
                         RegistryEntry::CurrentThread(_) => None,
                         RegistryEntry::BackgroundThread(_control_addr, thread_handle) => {
                             if thread_handle.thread().id() == current_thread.id() {
+                                debug!(
+                                    "[{}] [{}] skipping join of the current thread: {}",
+                                    self.name, i, actor_name
+                                );
                                 return None;
                             }
```

Comment:
I understand your confusion, but this is even more confusing, because if current thread is handled here, what does the `RegistryEntry::CurrentThread(_) => None,` arm do?

Reply:
I understand your confusion. :) `RegistryEntry::CurrentThread` is used when one does `system...run_and_block()` which runs the actor on the thread that called it. However this is a `SystemHande::shutdown()` method that can be called from a different thread.

I'll add a debug print to both places and avoid the word "current thread".

Reply:
> So, the intent is to skip joining the thread that called `shutdown()`?

Yup! Otherwise it would deadlock I think (you'd be trying to join your own thread). See 32be2a5089 for updated logging.